service/controller/clusterapi/v27/cloudconfig/legacy.go

```diff
@@ -1,18 +1,110 @@
 package cloudconfig
 
 import (
+	"net"
+	"strings"
+
 	g8sv1alpha1 "github.com/giantswarm/apiextensions/pkg/apis/provider/v1alpha1"
 	cmav1alpha1 "sigs.k8s.io/cluster-api/pkg/apis/cluster/v1alpha1"
+
+	"github.com/giantswarm/aws-operator/service/controller/clusterapi/v27/key"
 )
 
-func cmaClusterToG8sCluster(cr cmav1alpha1.Cluster) g8sv1alpha1.Cluster {
-	return g8sv1alpha1.Cluster{
-		// TODO
+func (c *CloudConfig) cmaClusterToG8sConfig(cr cmav1alpha1.Cluster) g8sv1alpha1.AWSConfigSpec {
+	return g8sv1alpha1.AWSConfigSpec{
+		Cluster: g8sv1alpha1.Cluster{
+			Calico: g8sv1alpha1.ClusterCalico{
+				CIDR:   c.calicoCIDR,
+				MTU:    c.calicoMTU,
+				Subnet: c.calicoSubnet,
```

Comment:
All of this appears to be sufficient for generating the cloud config. We still have to properly verify this later as we cannot really test it right now without the Cluster API types.